components/tikv_alloc/src/lib.rs

```diff
@@ -92,222 +93,26 @@
-                let mut prof = false;
-                let res = unsafe { jemallocator::mallctl_fetch(OPT_PROF, &mut prof) };
-                match res {
-                    Err(e) => {
-                        // Shouldn't be possible since mem-profiling is set
-                        panic!("is_profilng_on: {:?}", e);
-                    }
-                    Ok(_) => prof,
-                }
-            }
-
-            // Only trigger this test with jemallocs `opt.prof` set to
-            // true ala `MALLOC_CONF="prof:true"`. It can be run by
-            // passing `-- --ignored` to `cargo test -p tikv_alloc`.
-            //
-            // TODO: could probably unignore this by running a second
-            // copy of the executable with MALLOC_CONF set.
-            //
-            // TODO: need a test for the dump_prof(None) case, but
-            // the cleanup afterward is not simple.
-            #[test]
-            #[ignore]
-            fn test_profiling_memory() {
-                // Make sure somebody has turned on profiling
-                assert!(is_profiling_on(), r#"Set MALLOC_CONF="prof:true""#);
-
-                let dir = TempDir::new("test_profiling_memory").unwrap();
-
-                let os_path = dir.path().to_path_buf().join("test1.dump").into_os_string();
-                let path = os_path.into_string().unwrap();
-                super::dump_prof(Some(&path));
-
-                let os_path = dir.path().to_path_buf().join("test2.dump").into_os_string();
-                let path = os_path.into_string().unwrap();
-                super::dump_prof(Some(&path));
-
-                let files = fs::read_dir(dir.path()).unwrap().count();
-                assert_eq!(files, 2);
-
-                // Find the created files and check properties that
-                // indicate they contain something interesting
-                let mut prof_count = 0;
-                for file_entry in fs::read_dir(dir.path()).unwrap() {
-                    let file_entry = file_entry.unwrap();
-                    let path = file_entry.path().to_str().unwrap().to_owned();
-                    if path.contains("test1.dump") || path.contains("test2.dump") {
-                        let metadata = file_entry.metadata().unwrap();
-                        let file_len = metadata.len();
-                        assert!(file_len > 10); // arbitrary number
-                        prof_count += 1
-                    }
-                }
-                assert!(prof_count == 2);
-            }
-        }
-    }
-
-    #[cfg(not(feature = "mem-profiling"))]
-    mod profiling {
-        pub fn dump_prof(_path: Option<&str>) {}
-    }
-}
-
-#[cfg(not(all(unix, not(fuzzing), not(feature = "no-jemalloc"))))]
-mod imp {
-
-    use super::AllocStats;
-    use std::io;
+// The Jemallocator Allocator
+#[cfg(all(unix, not(fuzzing), feature = "jemalloc"))]
+#[global_allocator]
+static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
```

Comment:
Can the details be hidden inside module imp?

Reply:
Can't put in, don't work

Reply:
How about
```
mod imp {
    pub type Allocator = jemallocator::Jemalloc;
    pub const fn allocator() -> Allocator {
        jemallocator::Jemalloc
    }
}

static ALLOC: imp::Allocator = imp::allocator();
```
?

Reply:
That's a nice pattern.